internal/agent/agent.go

```diff
@@ -92,6 +93,20 @@ func (a *Agent) Run(ctx context.Context) error {
 		return err
 	}
 
+	var tpmClient *lifecycle.TpmClient
+	experimentalFeatures := experimental.NewFeatures()
+
+	if experimentalFeatures.IsEnabled() {
+		a.log.Warn("experimental features enabled: creating TPM client")
+		tpmClient, err = lifecycle.NewTpmClient(a.log)
+		if err != nil {
+			a.log.Infof("tpm is not available: %v", err)
+			tpmClient = nil
```

Comment:
if NewTpmClient returns non nil error tpmClient should always be nil.

Reply:
Reasonable. Fixed